infrastructure/balsa_queue/bq_main_macro.hh

```diff
@@ -12,39 +13,58 @@ void signal_handler(int s) {
-    struct sigaction sigIntHandler;                                           \
-    sigIntHandler.sa_handler = signal_handler;                                \
-    sigemptyset(&sigIntHandler.sa_mask);                                      \
-    sigIntHandler.sa_flags = 0;                                               \
-    sigaction(SIGINT, &sigIntHandler, NULL);                                  \
-    YAML::Node config;                                                        \
-    if (argc > 1) {                                                           \
-      try {                                                                   \
-        config = YAML::LoadFile(argv[1]);                                     \
-      } catch (YAML::BadFile e) {                                             \
-        std::cerr << "Could not find YAML file " << std::endl;                \
-        config = YAML::Node();                                                \
-      }                                                                       \
-    }                                                                         \
-    bq_type balsa_queue = bq_type();                                          \
-    if (config["bq_name"]) {                                                  \
-      balsa_queue.set_name(config["bq_name"].as<std::string>());              \
-      balsa_queue.gonogo().setName(config["bq_name"].as<std::string>());      \
-    } else {                                                                  \
-      balsa_queue.set_name(#bq_type);                                         \
-      balsa_queue.gonogo().setName(#bq_type);                                 \
-    }                                                                         \
-    balsa_queue.gonogo().nogo("init");                                        \
-    balsa_queue.set_comms_factory(std::make_unique<jet::MqttCommsFactory>()); \
-    balsa_queue.base_init();                                                  \
-    balsa_queue.init(config);                                                 \
-    while (!shutdown) {                                                       \
-      balsa_queue.base_loop();                                                \
-      balsa_queue.loop();                                                     \
-      usleep(balsa_queue.loop_delay_microseconds);                            \
-    }                                                                         \
-    balsa_queue.shutdown();                                                   \
-    return 0;                                                                 \
+void set_up_signal_handler() {
+  struct sigaction sigIntHandler;
+  sigIntHandler.sa_handler = signal_handler;
+  sigemptyset(&sigIntHandler.sa_mask);
+  sigIntHandler.sa_flags = 0;
+  sigaction(SIGINT, &sigIntHandler, NULL);
+}
+
+std::string parse_commandline_arguments(int argc, char* argv[]) {
```

Comment:
[taste] Recommend optional instead of empty-string to communicate no arguments

Reply:
:+1: